R/tax_unique.R

```diff
@@ -135,105 +132,55 @@ tax_unique <- function(occdf = NULL, binomial = NULL, species = NULL,
   if (!is.null(binomial) && !(binomial %in% colnames(occdf))) {
     stop("`occdf` does not contain column name provided to `binomial`")
   }
-  if (!is.null(binomial)) {
-    colnames(occdf)[colnames(occdf) == binomial] <-
-      "binomial"
-  }
 
   if (!is.null(species) && !(species %in% colnames(occdf))) {
     stop("`occdf` does not contain column name provided to `species`")
   }
-  if (!is.null(species)) {
-    colnames(occdf)[colnames(occdf) == species] <-
-      "species"
-  }
 
   if (!is.null(genus) && !(genus %in% colnames(occdf))) {
     stop("`occdf` does not contain column name provided to `genus`")
   }
-  if (!is.null(genus)) {
-    colnames(occdf)[colnames(occdf) == genus] <-
-      "genus"
-  }
 
-  if (is.null(family)) {
-    stop("`family` argument is missing. Provide column name of family names.")
-  }
-  if (!(family %in% colnames(occdf))) {
-    stop("`occdf` does not contain column name provided to `family`")
+  higher_args <- list(...)
+  higher_names <- names(higher_args)
+  higher_cols <- unname(unlist(higher_args))
+  if (length(higher_args) == 0) {
+    stop("At least one higher taxonomic level must be supplied (e.g. family)")
   }
-  colnames(occdf)[colnames(occdf) == family] <- "family"
-
-  if (!is.null(order) && !(order %in% colnames(occdf))) {
-    stop("`occdf` does not contain column name provided to `order`")
-  }
-  if (!is.null(order)) {
-    colnames(occdf)[colnames(occdf) == order] <-
-      "order"
-  }
-
-  if (!is.null(class) && !(class %in% colnames(occdf))) {
-    stop("`occdf` does not contain column name provided to `class`")
-  }
-  if (!is.null(class)) {
-    colnames(occdf)[colnames(occdf) == class] <-
-      "class"
+  for (level_label in names(higher_args)) {
```

Comment:
```suggestion
  for (level_label in higher_names) {
```
Couldn't this just be replaced with higher_names from L145?